tests/ui/layout/zero-sized-array-enum-niche.stderr

```diff
@@ -1,4 +1,4 @@
-error: layout_of(std::result::Result<[u32; 0], bool>) = Layout {
```

Comment:
This diff shows that `Layout` debug printing is affected. I think that's a problem, it makes it much harder to figure out which type a given layout is for (and staring at this debug output us not uncommon when debugging MIR/Miri issues).